### DISCUSSION TITLE
Any way to generate migration files that explain drift?

### DISCUSSION THREAD

**Discussion by katangafor** [prisma/prisma#20214] [Q&A] [answered]
Sometimes I or my coworkers will manually apply changes to the database. Some people on the team don't even have a typescript environment where they can use prisma: they work on a python project that accesses the same database. Of course this results in "drift detected". 

I tried using `prisma migrate diff` to generate an SQL script that could cover the difference from my migration history to my database structure, but I can't figure out how to make prisma recognize that SQL script as a migration. 

Even if I put the script in a folder in prisma/migrations, prisma doesn't recognize it as a migration. That's probably because there's nothing about it in the _prisma_migrations table, but I can't figure out how to _put_ a reference to the new migrations in the _prisma_migrations table. 

Is there any way to generate (or manually add) a migration to the migration history to explain schema drift?

**Comment by janpio** [comment, marked as answer]
I think you are looking for `prisma migrate resolve --applied`: https://www.prisma.io/docs/reference/api-reference/command-reference#migrate-resolve That will make sure that the migration is recorded as "applied" in `_prisma_migrations`, and will be used when you run `migrate dev` or a similar command again to calculate the diff between your database and your migration history.

**Reply by katangafor** [reply]
Okay that makes a lot of sense! It's pretty much the same process as baselining, then?
Thanks for the help!

**Reply by janpio** [reply]
Yes, baselining is the same when starting with a fully empty history - so the SQL is _everything_ and then the only migration that is marked as applied - and here it is only the diff, and then marking that applied on top of what already exists. Both has the goal that the full database schema is represented with migration files that are recorded in `_prisma_migrations`.

**Reply by katangafor** [reply]
Ok gotchya, thanks so much! Very very cool stuff.
